### ISSUE TITLE
Build failure: mangohud

### ISSUE THREAD

**Issue by dunk1452** [NixOS/nixpkgs#226001]
### Steps To Reproduce

1. build mangohud on nixos-unstable channel

### Build log

https://gist.github.com/dunk1452/6d0f1edef6b79b7e4ab3cf5789fa8c08

### Additional context

Qemu Virtual Environment, updated my flake and resulted in below.

```
building the system configuration...
error: builder for '/nix/store/v3hjcx3zkk0jafyrdrz4wl0yhziilahf-mangohud-0.6.8.drv' failed with exit code 1;
       last 10 log lines:
       > [39/66] Compiling C++ object src/libMangoHud.so.p/loaders_loader_nvml.cpp.o
       > [40/66] Compiling C++ object src/libMangoHud.so.p/nvml.cpp.o
       > [41/66] Compiling C++ object subprojects/imgui-1.81/libimgui.a.p/imgui.cpp.o
       > [42/66] Compiling C++ object src/libMangoHud.so.p/cpu.cpp.o
       > [43/66] Compiling C++ object src/libMangoHud.so.p/notify.cpp.o
       > [44/66] Compiling C++ object src/libMangoHud.so.p/battery.cpp.o
       > [45/66] Compiling C++ object subprojects/spdlog-1.8.5/src/libspdlog.a.p/spdlog.cpp.o
       > [46/66] Compiling C++ object subprojects/spdlog-1.8.5/src/libspdlog.a.p/fmt.cpp.o
       > [47/66] Compiling C++ object src/libMangoHud.so.p/gamepad.cpp.o
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/v3hjcx3zkk0jafyrdrz4wl0yhziilahf-mangohud-0.6.8.drv'.
error: 1 dependencies of derivation '/nix/store/w35gfg097dgchag7i2axa30jy49vn138-mangohud-0.6.8.drv' failed to build
error: 1 dependencies of derivation '/nix/store/x99d7d052516s4f1gxvyli2csd20ls0n-heroic-usr-target.drv' failed to build
error: 1 dependencies of derivation '/nix/store/lsq1afyhfaw0sq1nmakpiwiln4p55pyg-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/bshlcxypj34smfbbnlk2kg953ah3fqh6-heroic-fhs.drv' failed to build
error: 1 dependencies of derivation '/nix/store/iarbl4xqxdlbc0xq4qvmnb0m5y100j69-home-manager-generation.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qiqz6cpixj58qn1rkiqiwb6wsh7qcs4w-user-environment.drv' failed to build
error: 1 dependencies of derivation '/nix/store/h5rny3icr42yi776l7jv7dpqrcxih330-etc.drv' failed to build
error (ignored): error: cannot unlink '/tmp/nix-build-nvidia-x11-530.41.03-6.1.23.drv-1/kernel/nvidia': Directory not empty
error: 1 dependencies of derivation '/nix/store/cbdrhf1zdbrbpvjna3kp02w3pirl7a1h-heroic-bwrap.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9lnsjn2mrxvpx7jyffhzhy0bxjrsqqrf-nixos-system-nixos-23.05.20230412.fe2ecaf.drv' failed to build
```

### Notify maintainers

@kira-bruneau @zeratax 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.2.9-zen1, NixOS, 23.05 (Stoat), 23.05.20230410.db24d86`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(user): `"nixos"`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by kira-bruneau** [comment]
Duplicates #225986, and should also be fixed now with #225892!

https://hydra.nixos.org/job/nixpkgs/trunk/newsflash.x86_64-linux

**Comment by kira-bruneau** [comment]
Fixed in #225985
